editor/src/components/canvas/remix/remix-error-handling.test-utils.tsx

```diff
@@ -0,0 +1,128 @@
+  const renderResult = await renderTestEditorWithModel(
+    project,
+    'await-first-dom-report',
+    DefaultStartingFeatureSwitches,
+    undefined,
+    RegisteredCanvasStrategies,
+    notLoggedIn,
+    false,
+  )
+  await renderResult.dispatch([runDOMWalker()], true)
+  return renderResult
+}
+
+async function clickLinkWithTestId(editor: EditorRenderResult, testId: string) {
+  const targetElement = editor.renderedDOM.queryAllByTestId(testId)[0]
+  const targetElementBounds = targetElement.getBoundingClientRect()
+
+  await mouseClickAtPoint(targetElement, {
+    x: targetElementBounds.x + 5,
+    y: targetElementBounds.y + 5,
+  })
+}
+
+const CustomBoundaryText = 'I am a custom error boundary'
+const ErrorText = 'CATCH ME IF YOU CAN!'
+
+function createProject(withBoundary: boolean) {
+  return createModifiedProject({
+    [StoryboardFilePath]: `import * as React from 'react'
+    import { RemixScene, Storyboard } from 'utopia-api'
+    
+    export var storyboard = (
+      <Storyboard>
+        <RemixScene
+          style={{
+            width: 700,
+            height: 759,
+            position: 'absolute',
+            left: 212,
+            top: 128,
+          }}
+        />
+      </Storyboard>
+    )
+    `,
+    ['/src/root.js']: `import React from 'react'
+    import { Outlet } from '@remix-run/react'
+    
+    export default function Root() {
+      return (
+        <div>
+          <Outlet />
+        </div>
+      )
+    }
+    `,
+    ['/src/routes/_index.js']: `import React from 'react'
+    import { Link } from '@remix-run/react'
+    
+    export default function Index() {
+      return (
+        <Link
+          style={{
+            width: 100,
+            height: 100
+          }}
+          to='/thrower'
+          data-testid='remix-link'
+        >
+          Throw!
+        </Link>
+      )
+    }
+    `,
+    ['/src/routes/thrower.js']: `import React from 'react'
+    import { useRouteError } from '@remix-run/react'
+    
+    ${
+      withBoundary
+        ? `
+      export function ErrorBoundary() {
+        const error = useRouteError()
+        return <div>${CustomBoundaryText}</div>
+      }
+      `
+        : ``
+    }
+    
+    export default function Thrower() {
+      throw new Error('${ErrorText}')
+      return <h1>I won't render</h1>
+    }
+    `,
+  })
+}
+
+export async function runTestReturningErrorBoundaries(
+  withCustomBoundary: 'with-custom-boundary' | 'without-custom-boundary',
+): Promise<{ customBoundary: HTMLElement | undefined; canvasOverlay: HTMLElement | undefined }> {
+  const project = createProject(withCustomBoundary === 'with-custom-boundary')
+
+  const renderResult = await renderRemixProject(project)
+
+  await renderResult.dispatch([switchEditorMode(EditorModes.liveMode())], true)
+  await clickLinkWithTestId(renderResult, 'remix-link')
+
+  const [customBoundary] = renderResult.renderedDOM.queryAllByText(CustomBoundaryText)
```

Comment:
Should these allow more than one result?

Reply:
Nope! Changing them to `queryByText`